src/django_mysql/models/fields/datetime.py

```diff
@@ -0,0 +1,78 @@
+from django.core import checks
+from django.db.models import DateTimeField as DjangoDateTimeField
+from django.db.models.fields import DateTimeCheckMixin as DjangoDateTimeCheckMixin
+from django.utils import timezone
+
+
+class DateTimeCheckMixin(DjangoDateTimeCheckMixin):
+    def _check_mutually_exclusive_options(self):
+        # auto_now, on_upd and default are mutually exclusive
+        # options. The use of more than one of these options together
+        # will trigger an Error
+        # and django_mysql.models.DateTimeField should not be enabled 'auto_now_add=True'
+        mutually_exclusive_options = [
+            self.auto_now,
+            self.on_update_current_timestamp,
+            self.has_default(),
+        ]
+        enabled_options = [
+            option not in (None, False) for option in mutually_exclusive_options
+        ].count(True)
+        if enabled_options > 1:
+            return [
+                checks.Error(
+                    "The options auto_now, on_update_current_timestamp, and default "
+                    "are mutually exclusive. Only one of these options "
+                    "may be present. "
+                    "Recommend 'on_update_current_timestamp=True' ",
+                    obj=self,
+                    id="fields.E160",
+                )
+            ]
+        elif self.auto_now_add:
+            return [
+                checks.Error(
+                    "The options auto_now_add would not use with django_mysql.models.DateTimeField "
+                    "fix 'auto_now_add=True' to 'auto_now_add=False' "
+                    "or Use 'auto_now_add=True' with django.db.models.DateTimeField. ",
+                    obj=self,
+                    id="fields.E161",
+                )
+            ]
+        else:
+            return []
+
+
+class DateTimeField(DateTimeCheckMixin, DjangoDateTimeField):
+    def __init__(self, on_update_current_timestamp=False, **kwargs):
+        self.on_update_current_timestamp = on_update_current_timestamp
+        super(DateTimeField, self).__init__(**kwargs)
+
+    def db_type_suffix(self, connection):
+        db_type_suffix = super().db_type_suffix(connection)
+        if db_type_suffix is None:
+            db_type_suffix = "ON UPDATE CURRENT_TIMESTAMP(6)"
+        else:
+            db_type_suffix += "ON UPDATE CURRENT_TIMESTAMP(6)"
+        return db_type_suffix
+
+    def pre_save(self, model_instance, add):
+        if self.auto_now or self.on_update_current_timestamp:
+            value = timezone.now()
+            setattr(model_instance, self.attname, value)
```

Comment:
it's a bit disingenous to set the attribute to the current time in python when it's updated in the database, because the two clocks may be out of sync.

Reply:
I intended .save() depend on  **current time in python**
and QuerySet.update() depend on  **MYSQL current_timestamp** due to ``` on_update_current_timestamp=True ```


#### because
we can not control ```field.defualt``` Option in Django Field  & Update SQL 
it's controlled in ``` django.db.backend.*.schemas.XXXDatabaseSchemaEditor```
and
Update SQL  should ignore `DateTImeField(on_update_current_timestamp=True)`
When UPDATE SQL execution

we need to  override `django MYSQLDatabaseSchemaEditor`
https://github.com/django/django/blob/master/django/db/backends/mysql/schema.py
And django.QuerySet.update()

but I thought  it's too huge change Just to modify one little thing(DateTimeField)

I agree `it's a bit disingenous` 
but  I try to make coherent action
` .save()    depend on  current time in python`
`.update() depend on  MYSQL current_timestamp`

With my short insight
If  develop DateTimeField with Perfectly depended MYSQL current_timestamp
We need to refactor Django Backend schemas.DatabaseSchemaEditor
 that  DDL is depended

Reply:
I don't see what DDL needs modifying for this?

Reply:
I'll think about it more.